Sources_v3/Database/DTOs/UserDTO_Tests.swift

```diff
@@ -174,4 +174,116 @@ class UserDTO_Tests: XCTestCase {
         // Check the online status is `false`
         XCTAssertEqual(database.viewContext.user(id: userId)?.isOnline, false)
     }
+    
+    func test_userWithUserListQuery_isSavedAndLoaded() {
+        let query = UserListQuery(filter: .autocomplete("name", with: "a"))
+        
+        // Create two users
+        let payload1 = dummyUser
+        let id1 = payload1.id
+        
+        let payload2 = dummyUser
+        
+        // Save the channels to DB, but only user 1 is associated with the query
+        database.write { session in
```

Comment:
You can change this to `writeSynchronously` and the assertion below can become synchronous